config/src/main/java/com/yahoo/vespa/config/JRTConnectionPool.java

```diff
@@ -67,20 +65,34 @@ public synchronized JRTConnection getCurrent() {
         return currentConnection;
     }
 
-    /**
-     * Returns and set the current JRTConnection instance by randomly choosing
-     * from the available sources (this means that you might end up using
-     * the same connection).
-     *
-     * @return a JRTConnection
-     */
-    public synchronized JRTConnection setNewCurrentConnection() {
+    @Override
+    public synchronized JRTConnection switchConnection() {
         List<JRTConnection> sources = getSources();
-        currentConnection = sources.get(ThreadLocalRandom.current().nextInt(0, sources.size()));
+        if (sources.size() <= 1) return currentConnection;
+
+        List<JRTConnection> healthySources = sources.stream()
```

Comment:
This selects the healthy connections.  But isn't the healthiness of a connection only updated when it is "current"?  And so what may happen is:
* Say there are 3 members A, B, and C.  They start out as healthy, and A is chosen to be current.
* Say A becomes unhealhty, and then a random of B and C is chosen, say B.
* While B is the current connection, A has no chance of changing its healthiness so it stays unhealthy.
* When B becomes unhealthy, C is chosen.
* When C becomes unhealthy, C is kept per below test on `size() == 0`

Reply:
Yes, you are right. We could either start doing ping out-of-band or try an unhealthy connection after some time (if there is not any healthy ones to choose from). Will look into it.

Reply:
setNewCurrentConnection() is only called on creation or when exactly 1 connection is bad.  So a simple method might be to select one randomly, or select randomly except that one connection, respectively?

Reply:
Thanks, I have implemented a suggestion with some test